gcc/rust/ast/rust-ast-full-test.cc

```diff
@@ -4120,6 +4122,37 @@ Module::get_filename ()
 			: current_directory_name + expected_dir_path;
 }
 
+void
+Module::load_items ()
+{
+  std::string mod_file = get_filename ();
+
+  // We will already have errored out appropriately in the get_filename ()
+  // method
+  if (mod_file.empty ())
+    return;
+
+  RAIIFile file_wrap (mod_file.c_str ());
+  Linemap *linemap = Session::get_instance ().linemap;
+
+  if (file_wrap.get_raw () == nullptr)
+    rust_fatal_error (Location (), "cannot open module file %s: %m",
+		      mod_file.c_str ());
+
+  rust_debug ("Attempting to parse file %s", mod_file.c_str ());
+
+  Lexer lex (mod_file.c_str (), std::move (file_wrap), linemap);
+  Parser<Lexer> parser (std::move (lex));
+
+  auto items = parser.parse_items ();
+
+  for (const auto &error : parser.get_errors ())
+    error.emit_error ();
+
+  items = std::move (items);
+  kind = ModuleKind::LOADED;
```

Comment:
Can you investigate to see if rustc has a ModuleKind::ERROR for the case where we fail to load? I am not sure if it actually helps with anything. Even if they do have this state lets not add it to this PR.

Reply:
So the Rust enum is the following:
```rust
enum ModKind {
    Load(Vec<Pointer<Item>>, Inline, Span),
    Unloaded
}
```
So they do not have an `Error` variant. The enum we use in gccrs is to try and reproduce those "enum variants" since we can't have them in C++ sadly. So we always keep a vector, which might be empty, and then the `kind` which corresponds to `Loaded` or `Unloaded`